connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/table/BigtableTableTest.java

```diff
@@ -0,0 +1,252 @@
+ * limitations under the License.
+ */
+
+package com.google.flink.connector.gcp.bigtable.table;
+
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.catalog.Column;
+import org.apache.flink.table.catalog.ResolvedSchema;
+import org.apache.flink.table.catalog.UniqueConstraint;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.factories.utils.FactoryMocks;
+
+import com.google.flink.connector.gcp.bigtable.table.config.BigtableConnectorOptions;
+import com.google.flink.connector.gcp.bigtable.testingutils.TestingUtils;
+import com.google.flink.connector.gcp.bigtable.utils.ErrorMessages;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Unit tests for the {@link BigtableDynamicTableSink} class.
+ *
+ * <p>This class verifies the functionality of the {@link BigtableDynamicTableSink} by testing its
+ * ability to create a Bigtable sink with different configurations, including column family mode and
+ * nested rows mode. It also includes tests for compatibility checks and row key validation.
+ */
+public class BigtableTableTest {
+    private static final List<Column> SCHEMA_LIST =
+            Arrays.asList(
+                    Column.physical(TestingUtils.ROW_KEY_FIELD, DataTypes.STRING().notNull()),
+                    Column.physical(TestingUtils.STRING_FIELD, DataTypes.STRING()),
+                    Column.physical(TestingUtils.INTEGER_FIELD, DataTypes.INT()));
+
+    private static final List<Column> NESTED_SCHEMA_LIST =
+            Arrays.asList(
+                    Column.physical(TestingUtils.ROW_KEY_FIELD, DataTypes.STRING().notNull()),
+                    Column.physical(
+                            TestingUtils.NESTED_COLUMN_FAMILY_1,
+                            DataTypes.ROW(
+                                    DataTypes.FIELD(TestingUtils.STRING_FIELD, DataTypes.STRING()),
+                                    DataTypes.FIELD(TestingUtils.INTEGER_FIELD, DataTypes.INT()))),
+                    Column.physical(
+                            TestingUtils.NESTED_COLUMN_FAMILY_2,
+                            DataTypes.ROW(
+                                    DataTypes.FIELD(
+                                            TestingUtils.STRING_FIELD_2, DataTypes.STRING()))));
+
+    @Test
+    public void testDynamicTableBigtableSink() throws IOException {
```

Comment:
we should also add tests to validate errors are throw when required options arent set correctly

Reply:
I didn't know this was possible since it was a Flink native error (helper.validate()), I was about to give up but managed to do it. Thanks!